angular/src/app/interfaces/user.ts

```diff
@@ -0,0 +1,12 @@
+export interface Credential {
```

Comment:
Move the interfaces into "src/shared/interfaces.ts"

We should separate all the interfaces in individual files later, but for now, put them there.